### PR TITLE
N8n 2582 plus endpoint

### PR DESCRIPTION
**Pull request by mutdmour**

### CODE REVIEW

File: packages/editor-ui/src/plugins/PlusEndpointType.js [hunk 1]
```diff
@@ -0,0 +1,495 @@
...
+		const message = container.querySelector('.drop-hover-message');
+		const plusStalk = this.canvas.querySelector('.plus-stalk');
+		const successOutput = this.canvas.querySelector('.plus-stalk span');
+
+		this.setSuccessOutput = (label) => {
+			this.canvas.classList.add('success');
+			if (this.showOutputLabel) {
+				successOutput.textContent = label;
+				this.label = label;
+				this.labelOffset = successOutput.offsetWidth;
```
Comment (mutdmour) [inline packages/editor-ui/src/plugins/PlusEndpointType.js:395]: increase to accommodate longer labels

**Review by mutdmour — COMMENTED** [review]
hints

File: packages/editor-ui/src/store.ts [hunk 2]
```diff
@@ -862,7 +862,7 @@ export const store = new Vuex.Store({
 			return state.workflowExecutionData;
 		},
 		getWorkflowRunData: (state): IRunData | null => {
-			if (state.workflowExecutionData === null) {
+			if (!state.workflowExecutionData || !state.workflowExecutionData.data || !state.workflowExecutionData.data.resultData) {
```
Comment (BHesseldieck) [inline packages/editor-ui/src/store.ts:865]: we need to get the optional chaining to the FE xD

File: packages/editor-ui/src/views/NodeView.vue [hunk 3]
```diff
@@ -360,7 +361,7 @@ export default mixins(
 				this.$externalHooks().run('execution.open', { workflowId: data.workflowData.id, workflowName: data.workflowData.name, executionId });
 				this.$telemetry.track('User opened read-only execution', { workflow_id: data.workflowData.id, execution_mode: data.mode, execution_finished: data.finished });
 
-				if (data.finished !== true && data.data.resultData.error) {
+				if (data.finished !== true && data && data.data && data.data.resultData && data.data.resultData.error) {
```
[the remark this answers is not in the document]
Comment (BHesseldieck) [reply packages/editor-ui/src/views/NodeView.vue:364]: I want optional chaining 😩

File: packages/editor-ui/src/views/NodeView.vue [hunk 4]
```diff
@@ -1707,8 +1718,13 @@ export default mixins(
 				// it visibly stays behind free floating without a connection.
 				connection.removeOverlays();
 
+				const sourceEndpoint = connection.endpoints && connection.endpoints.length && connection.endpoints[0];
```
Comment (BHesseldieck) [inline packages/editor-ui/src/views/NodeView.vue:1721]: isn't this sufficient? Empty array would also mean its undefined/false
```suggestion
				const sourceEndpoint = connection.endpoints && connection.endpoints[0];
```

File: packages/editor-ui/src/views/canvasHelpers.ts [hunk 5]
```diff
@@ -511,20 +499,32 @@ export const getOutputSummary = (data: ITaskData[], nodeConnections: NodeInputCo
...
+			if (!outputMap[sourceOutputIndex][NODE_OUTPUT_DEFAULT_KEY]) {
+				outputMap[sourceOutputIndex][NODE_OUTPUT_DEFAULT_KEY] = {};
+				outputMap[sourceOutputIndex][NODE_OUTPUT_DEFAULT_KEY][0] = {
+					total: 0,
+					iterations: 0,
+				};
+			}
+
+			outputMap[sourceOutputIndex][NODE_OUTPUT_DEFAULT_KEY][0].total += output ? output.length : 0;
+			outputMap[sourceOutputIndex][NODE_OUTPUT_DEFAULT_KEY][0].iterations += output ? 1 : 0;
```
Comment (BHesseldieck) [inline packages/editor-ui/src/views/canvasHelpers.ts:517]: maybe assigning `outputMap[sourceOutputIndex][NODE_OUTPUT_DEFAULT_KEY][0]` to a temporary variable for readability

**Review by BHesseldieck — APPROVED** [review]
great work! 💯 

just some small comments/details

File: packages/editor-ui/src/store.ts (hunk 2 above)
Comment (mutdmour) [reply packages/editor-ui/src/store.ts:865]: yes

File: packages/editor-ui/src/views/NodeView.vue (hunk 4 above)
Comment (mutdmour) [reply packages/editor-ui/src/views/NodeView.vue:1721]: ya I guess it's the same

Comment (mutdmour) [reply packages/editor-ui/src/views/NodeView.vue:1721]: updated

File: packages/editor-ui/src/views/canvasHelpers.ts (hunk 5 above)
Comment (mutdmour) [reply packages/editor-ui/src/views/canvasHelpers.ts]: updated

**Comment by janober** [comment]
Great work. Got merged.

**Comment by janober** [comment]
Got released with `n8n@0.152.0`

**Pull request merged**